spec/tapioca/cli_spec.rb

```diff
@@ -569,6 +572,95 @@ def body=(body); end
       refute_path_exists("#{outdir}/namespace/comment.rbi")
       refute_path_exists("#{outdir}/user.rbi")
     end
+
+    describe("verify") do
+      describe("with no changes") do
+        before do
+          execute("dsl")
+        end
+
+        it 'does nothing and returns exit_status 0' do
+          output = execute("dsl", "--verify")
+
+          assert_includes(output, <<~OUTPUT)
+            Nothing to do, all RBIs are up-to-date.
+          OUTPUT
+          assert_includes($?.to_s, "exit 0") # rubocop:disable Style/SpecialGlobalVars
+        end
+      end
+
+      describe("with new file") do
+        before do
+          execute("dsl")
+          File.write(repo_path / "lib" / "image.rb", <<~RUBY)
+            # typed: true
+            # frozen_string_literal: true
+
+            class Image
+              include(SmartProperties)
+
+              property :title, accepts: String
+            end
+          RUBY
+        end
+
+        after do
+          File.delete(repo_path / "lib" / "image.rb") if File.exist?(repo_path / "lib" / "image.rb")
+        end
+
+        it 'advises of new file(s) and returns exit_status 1' do
+          output = execute("dsl", "--verify")
+
+          assert_includes(output, <<~OUTPUT)
+            RBI files are out-of-date, please run `generate command` to update.
+            Reason: New file(s) introduced.
+          OUTPUT
+          assert_includes($?.to_s, "exit 1") # rubocop:disable Style/SpecialGlobalVars
```

Comment:
Want to make a special note about this. I was intending to use `Process.last_status` which wasn't available in Ruby 2.4. So I opted to use this with the `rubocop` exception as it's just for a test to work around the `IO.popen(...)`